DarkEmulator HaloMassFunction

Keitaro1020ishikawa opened the pull request:
I have implemented halo mass function calculated by Dark Emulator (Nishimichi et al. 2019).
The file name is nishimichi19.py.
Accordingly, import statement is also added into __init__.py file in hmfunc directory.
Any comments or questions are welcome.

HironaoMiyatake commented on pyccl/halos/hmfunc/nishimichi19.py:
```diff
@@ -0,0 +1,79 @@
+                slope = np.log(mfp[1]/mfp[0])/np.log(m0[1]/m0[0])
+                mfh[m_lo] = mfp[0]*(Mh[m_lo]/m0[0])**slope
+            else:
+                raise RuntimeError("Input mass range is not supported. The reliable support range is from 10^12 to 10^16 Mo/h. If you want to use unsupport range, please turn on input value of 'extrapolate'.")        
```
It would be better to add what happened when the `extrapolation` is turned on.

How about
Input mass range is not supported. The supported range is from 10^12 to 10^16 Msun/h. If you want to use an unsupported range, put 'extrapolate==True' in the input. Then this function extrapolates mass function outside or the supported range.

Keitaro1020ishikawa commented:
I have just added a benchmark test for the Dark Emulator halo mass function (test_nishimichi19_hmf.py, hmf_nishimichi19.txt).
I also changed some of the RuntimeError statements in /hmfunc/nishimichi19.py.

damonge reviewed (commented):
Just a  couple final comments.
